pkg/services/sqlstore/sqlstore.go

```diff
@@ -265,6 +266,32 @@ func (ss *SqlStore) getEngine() (*xorm.Engine, error) {
 	}
 
 	sqlog.Info("Connecting to DB", "dbtype", ss.dbCfg.Type)
+	if ss.dbCfg.Type == migrator.SQLITE && strings.HasPrefix(connectionString, "file:") {
+		exists, err := fs.Exists(ss.dbCfg.Path)
+		if err != nil {
+			return nil, errutil.Wrapf(err, "can't check for existence of %q", ss.dbCfg.Path)
+		}
+
+		const perms = 0640
+		if !exists {
+			ss.log.Info("Creating SQLite database file", "path", ss.dbCfg.Path)
+			f, err := os.OpenFile(ss.dbCfg.Path, os.O_CREATE|os.O_RDWR, perms)
+			if err != nil {
+				return nil, errutil.Wrapf(err, "failed to create SQLite database file %q", ss.dbCfg.Path)
+			}
+			f.Close()
```

Comment:
Should maybe check for the error on close and return it? I'm not sure if close already being called is the only possible error case or not.

Also maybe this can be its own function

Reply:
Good point @kylebrandt on reporting error on close. I actually avoided checking it on purpose, but realize in retrospect that a closing error here could mean it didn't get created.